core/variant.cpp

```diff
@@ -1574,6 +1574,16 @@ Variant::operator String() const {
 					};
 				};
 #endif
+				if (_get_obj().obj->has_method("_to_string")) {
+					Variant::CallError ce;
+					Variant ret = _get_obj().obj->call("_to_string", NULL, 0, ce);
+
+					// Make sure a Variant::String was returned
+					if (ce.error == Variant::CallError::CALL_OK && ret.get_type() == Variant::STRING) {
```

Comment:
@vnen I've made the changes here to make sure the function returned the right type.

Reply:
Tested with a release export also, if needed I can provide the project I used.